src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -647,6 +648,18 @@ void SettingsLoader::_parseFragment(const winrt::hstring& source, const std::str
             }
             CATCH_LOG()
         }
+
+        // Construct a temp Json::Value that contains ONLY the actions from the
+        // fragment. This will allow fragments to add actions, but not
+        // necessarily set other global properties.
+        Json::Value tmp = {};
+        tmp[ActionsKey.data()] = json.root[ActionsKey.data()];
+
+        // Now parse that tmep json object, as if it were a global settings
+        // blob. Manually opt-out of keybinding parsing - fragments shouldn't be
+        // allowed to bind actions to keys directly. We may want to revisit
+        // circa GH#2205
+        settings.globals->LayerJson(tmp, false);

```

Comment:
For this purpose I'd personally add a `GlobalAppSettings::LayerActionsFrom` method.